### ISSUE TITLE
Implement rate limiter on all endpoints to prevent misuse

### ISSUE THREAD

**Issue by shreyanshdwivedi** [fossasia/open-event-server#6037]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

Currently, there is no limiting factor on endpoints. Anyone can make a script and misuse it.

**Additional context**

https://github.com/fossasia/open-event-server/issues/6033 https://github.com/fossasia/open-event-server/issues/6033#issuecomment-501289551

**Comment by iamareebjamal** [comment]
Please implement rate limiting with user ID as key and fallback to IP for only unauthenticated users. I used to think this naive way of rate limiting is very easy until users of my college started to get limited despite being working from different computers. Because computers behind universities and companies sit behind a proxy which does Network Address Translation and maps internal IPs to a set of external IPs and blocking an IP would mean blocking the entire university and office. You would have implemented a denial of service for those users

**Comment by shreyanshdwivedi** [comment]
I agree with your concern. I'm currently reading the docs and will keep it in mind

**Comment by iamareebjamal** [comment]
Also, this'll work for app.route

Also see what can be done for jsonapi controllers
